includes/blocks/course-theme/class-course-progress-counter.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * File containing the Course_Progress_Counter class.
+ *
+ * @package sensei
+ * @since 3.13.4
+ */
+
+namespace Sensei\Blocks\Course_Theme;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use \Sensei_Blocks;
+
+/**
+ * Class Course_Progress_Counter is responsible for rendering the '1 of 10 lessons complete (10%)' block.
+ */
+class Course_Progress_Counter {
+	/**
+	 * Course_Progress_Counter constructor.
+	 */
+	public function __construct() {
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/course-theme-course-progress-counter',
+			[
+				'render_callback' => [ $this, 'render' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render() : string {
+		$course_id = \Sensei_Utils::get_current_course();
+		if ( ! $course_id ) {
+			return '';
+		}
+
+		$stats  = \Sensei()->course->get_progress_stats( $course_id );
+		$output = sprintf(
+			/* translators: Placeholder %1$d is the completed lessons count, %2$d is the total lessons count and %3$d is the percentage of completed lessons. */
+			__( '%1$d of %2$d lessons complete (%3$d%%)', 'sensei-lms' ),
+			$stats['completed_lessons_count'],
+			$stats['lessons_count'],
+			$stats['completed_lessons_percentage']
+		);
+
+		return ( "
+			<div class='sensei-course-theme-course-progress'>
```

Comment:
Not a critical thing, but since we use double quotes in all the HTML of the plugin, I think it would be nice to use double quotes here too. :)
The same applies to the other classes. ;)